.github/workflows/build-app.yml

```diff
@@ -4,106 +4,151 @@ on: [push]
 
 jobs:
   build-macos:
-    runs-on: macos-10.15
+    runs-on: macos-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - name: Get yarn cache
-        id: yarn-cache
-        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - uses: pnpm/action-setup@v2
+        name: Install pnpm
+        id: pnpm-install
+        with:
+          version: 8
```

Comment:
It's not as strong an argument, but you could store the pnpm version in `package.json`'s `packageManager` field: https://github.com/pnpm/action-setup#version.

Reply:
Sounds good to me! Less repeated code, even if it is in a `.yml`.